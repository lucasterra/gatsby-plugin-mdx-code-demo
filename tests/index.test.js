// @ts-nocheck
jest.mock('fs');
jest.mock('path');

const fs = require('fs');
const path = require('path');
const mdx = require('@mdx-js/mdx');
const plugin = require('../index');

// Mocks
const mdxText = `
# Title

[StaticAppBar](demo://static-appbar.js)

Lorem ipsum dolor sit amet.
`;

const demoCode = `
// prettier-ignore-start
import React from 'react'
// I wanna add some comments, just to make sure it will also escape them

const StaticAppBar = () => {
  return (
    <div>This is my appbar, it is so cool, right?</div>
  );
}

export default StaticAppBar
// prettier-ignore-end
`;

describe('gatsby-mdx-code-demo', () => {
  beforeEach(() => {
    fs.stat.mockReset();
    fs.stat.mockImplementation((file, cb) => {
      cb(null, { isFile: true, isMock: true });
    });

    fs.readFile.mockReset();
    fs.readFile.mockImplementation((file, openMode, cb) => {
      cb(null, demoCode);
    });

    path.dirname.mockReset();
    path.dirname.mockReturnValue('/path/to/some_directory');

    path.join.mockReset();
    path.join.mockReturnValue('/path/to/some_directory/static-appbar.js');
  });

  describe('include code demo', () => {
    let compiler;
    beforeAll(async() => {
      const demoPlugin = () => async (tree) => {
        await plugin(
          { markdownAST: tree, markdownNode: { fileAbsolutePath: __dirname } },
          { demoComponent: './demo-test' },
        );
      };

      compiler = (text) => mdx(text, { remarkPlugins: [[demoPlugin]] });
    });

    it('should work as expected', async () => {
      const textAST = await compiler(mdxText);
      expect(textAST).toMatchSnapshot();
    });
  });
});
