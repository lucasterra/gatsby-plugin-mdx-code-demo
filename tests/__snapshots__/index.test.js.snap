// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gatsby-mdx-code-demo include code demo should work as expected 1`] = `
"/* @jsx mdx */
import LocalComponent_a210b9b535d9e8dc491d91e896c765dc_0 from '/path/to/some_directory/.cache/code-demo-imports/LocalComponent_a210b9b535d9e8dc491d91e896c765dc_0.js'

const makeShortcode = name => function MDXDefaultShortcode(props) {
      console.warn(\\"Component \\" + name + \\" was not imported, exported, or provided by MDXProvider as global scope\\")
      return <div {...props}/>
    };
const CodeDemo = makeShortcode(\\"CodeDemo\\");
const layoutProps = {
  
};
const MDXLayout = \\"wrapper\\"
export default function MDXContent({
  components,
  ...props
}) {
  return <MDXLayout {...layoutProps} {...props} components={components} mdxType=\\"MDXLayout\\">

    <h1>{\`Title\`}</h1>
    <p><a parentName=\\"p\\" {...{
        \\"href\\": \\"demo://static-appbar.js\\"
      }}>{\`StaticAppBar\`}</a></p>
    <CodeDemo code={\\"// prettier-ignore-start\\\\nimport React from 'react'\\\\n// I wanna add some comments, just to make sure it will also escape them\\\\n\\\\nconst StaticAppBar = () => {\\\\n  return (\\\\n    <div>This is my appbar, it is so cool, right?</div>\\\\n  );\\\\n}\\\\n\\\\nexport default StaticAppBar\\\\n// prettier-ignore-end\\"} lang=\\"jsx\\" mdxType=\\"CodeDemo\\"><LocalComponent_a210b9b535d9e8dc491d91e896c765dc_0 mdxType=\\"LocalComponent_a210b9b535d9e8dc491d91e896c765dc_0\\" /></CodeDemo>
    <p>{\`Lorem ipsum dolor sit amet.\`}</p>
    </MDXLayout>;
}

;
MDXContent.isMDXComponent = true;"
`;
